packages/grid/_modules_/grid/models/colDef/gridDefaultColumnTypes.ts

```diff
@@ -2,16 +2,18 @@ import { GridColumnTypesRecord } from './gridColTypeDef';
 import { GRID_STRING_COL_DEF } from './gridStringColDef';
 import { GRID_NUMERIC_COL_DEF } from './gridNumericColDef';
 import { GRID_DATE_COL_DEF, GRID_DATETIME_COL_DEF } from './gridDateColDef';
+import { GRID_BOOLEAN_COL_DEF } from './gridBooleanColDef';
 
 export const DEFAULT_GRID_COL_TYPE_KEY = '__default__';
 export const getGridDefaultColumnTypes: () => GridColumnTypesRecord = () => {
   const nativeColumnTypes = {
-    string: { ...GRID_STRING_COL_DEF },
-    number: { ...GRID_NUMERIC_COL_DEF },
-    date: { ...GRID_DATE_COL_DEF },
-    dateTime: { ...GRID_DATETIME_COL_DEF },
+    string: GRID_STRING_COL_DEF,
```

Comment:
it looks nicer but it will create issues if ppl start to mutate the string type for example, as they will mutate the value for the whole app, all columns...

Murphy's law 🤔

Reply:
Where should we stop in this case, clone everything? I personally think that React developers understand that immutability is the norm. But if we anticipate this as a foot gun and to improve the DX, I think that can freeze deep the structure in dev mode so we don't have to pay the performance cost in production.

Reply:
> > By "visible" do you mean to show the border when the boolean field is in edit mode?
> 
> @m4theushw Yes, it's my proposal to fix this problem: the current UX is not great. Consider the following: false value. Can you easily tell if the cell is focused or in view mode, or in edit mode?

IMO, it should be a tick and a cross in view mode, and a checkbox in edit mode.
like the isFilled column of commodity
![image](https://user-images.githubusercontent.com/936978/114195447-c1a23a80-9950-11eb-8373-e16b8cca8ad1.png)

Reply:
the clone would only occur when the component mount. The deepFreeze would occur on state change on a quite big structure so it would have a perf hit. Not really in favor of that, as the cost outweighs the benefits.

Where should we stop in this case, clone everything? Well yes or we can use a function like I have done for filter operators. Better be safe than sorry 😄

Reply:
My biggest concern with the clone is that it allows having leaks to go unnoticed in our component.

Which performance hit are you referring to? A deep freeze on 100,000 rows costs about 400ms on my laptop. So I assume that the columns being at a x1000 smaller structure will cost ~1m, and **only** in development mode.

Reply:
try to do it with streaming, sorting and filtering in place